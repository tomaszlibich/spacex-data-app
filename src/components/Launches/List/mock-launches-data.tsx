export const mockLaunchData = {
    crew: null,
    details: null,
    flight_number: 0,
    is_tentative: false,
    launch_date_local: '2020-10-24T11:31:00-04:00',
    launch_date_unix: 1603553460,
    launch_date_utc: '2020-10-24T15:31:00.000Z',
    launch_site: {
        site_id: 'xyz',
        site_name: 'XYZ',
        site_name_long: 'XYZabc'
    },
    launch_success: true,
    launch_window: null,
    launch_year: '123',
    links: {
        article_link: null,
        flickr_images: [],
        mission_patch: '123456',
        mission_patch_small: '123',
        presskit: null,
        reddit_campaign: 'xyz',
        reddit_launch: 'abc',
        reddit_media: null,
        reddit_recovery: null,
        video_link: 'xyz',
        wikipedia: 'xyz',
    },
    mission_id: ['aaa'],
    mission_name: 'xyz',
    rocket: {
        rocket_id: 'xyz',
        rocket_name: 'xyz',
        rocket_type: 'abc',
        first_stage: {
            cores: []
        },
        second_stage: {
            block: 0,
            payloads: []
        }
    },
    ships: ['aaa'],
    static_fire_date_unix: 1,
    static_fire_date_utc: 2,
    tbd: false,
    telemetry: {
        flight_club: null
    },
    tentative_max_precision: '123',
    timeline: null
};

export const threeLaunchItems = [{
    ...mockLaunchData,
    flight_number: 1
}, {
    ...mockLaunchData,
    flight_number: 2
}, {
    ...mockLaunchData,
    flight_number: 3
}]