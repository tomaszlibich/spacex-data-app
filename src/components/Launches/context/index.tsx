import { createContext } from "react";
import { LaunchesContextType } from "../types";

export const initialData: LaunchesContextType = {
  launches: [],
  sorting: null,
  setSorting: () => {},
  search: undefined,
  setSearch: () => {},
};

export const LaunchesContext = createContext(initialData);
