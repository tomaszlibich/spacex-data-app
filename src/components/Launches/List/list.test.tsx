import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { LaunchesContext, initialData } from "../context";
import { LaunchesList } from "./index";
import { mockLaunchData, threeLaunchItems } from './mock-launches-data';

configure({ adapter: new Adapter() });

const initialContextData = { ...initialData };

let component;
let tree: any;
let wrapper: any;

describe("LaunchesList", () => {
  describe('When component is rendered', () => {
    beforeEach(() => {
      component = renderer.create(
        <LaunchesContext.Provider value={initialData}>
          <LaunchesList />
        </LaunchesContext.Provider>,

      );
      tree = component.toJSON();
    });

    it("should match the snapshot", () => {
      expect(tree).toMatchSnapshot();
    });

    describe('And there is no launch items available', () => {
      beforeEach(() => {
        wrapper = mount(
          <LaunchesContext.Provider value={{ ...initialContextData, launches: [] }}>
            <LaunchesList />
          </LaunchesContext.Provider>
        )
      });

      it("should render no items on the list", async () => {
        expect(wrapper.find('li').length).toEqual(0);
      });
    });

    describe('And there is only one launch item available', () => {
      beforeEach(() => {
        wrapper = mount(
          <LaunchesContext.Provider value={{ ...initialContextData, launches: [mockLaunchData] }}>
            <LaunchesList />
          </LaunchesContext.Provider>
        )
      });

      it("should render only one item on the list", async () => {
        expect(wrapper.find('li').length).toEqual(1);
      });
    });

    describe('And there are three launch items available', () => {
      beforeEach(() => {
        wrapper = mount(
          <LaunchesContext.Provider value={{ ...initialContextData, launches: threeLaunchItems }}>
            <LaunchesList />
          </LaunchesContext.Provider>
        )
      });

      it("should render three items on the list", async () => {
        expect(wrapper.find('li').length).toEqual(3);
      });
    });
  });
});
