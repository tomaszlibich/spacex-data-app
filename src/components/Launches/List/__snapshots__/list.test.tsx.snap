// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchesList When component is rendered should match the snapshot 1`] = `
<div
  className="list"
>
  <p
    className="noData"
  >
    No launch data available...
  </p>
</div>
`;
